[GLUTEN-8852][TEST] Support Spark 4.0 in gluten-it

PHILO-HE opened the pull request:
<!--
Thank you for submitting a pull request! Here are some tips:

1. For first-time contributors, please read our contributing guide:
   https://github.com/apache/incubator-gluten/blob/main/CONTRIBUTING.md
2. If necessary, create a GitHub issue for discussion beforehand to avoid duplicate work.
3. If the PR is specific to a single backend, include [VL] or [CH] in the PR title to indicate the
   Velox or ClickHouse backend, respectively.
4. If the PR is not ready for review, please mark it as a draft.
-->

## What changes are proposed in this pull request?

This PR makes the code under gluten-it compatible with Spark 4.0 and scala-2.13,
and also enables Spark-4.0 TPC-H test in CI.

Fixes #10438.

<!--
Provide a clear and concise description of the changes introduced in this PR.
Ensure the PR description aligns with the code changes, especially after updates.
If applicable, include "Fixes #<GitHub_Issue_ID>" to automatically close the corresponding issue
when the PR is merged.
-->

## How was this patch tested?
CI.

<!--
Describe how the changes were tested, if applicable.
Include new tests to validate the functionality, if necessary.
For UI-related changes, attach screenshots to demonstrate the updates.
-->

PHILO-HE commented:
@zhztheplayer, could you take a look? Thanks.

zhztheplayer reviewed (approved):
No idea how you got rid of the shims but thanks for iterating :)

PHILO-HE commented:
> No idea how you got rid of the shims but thanks for iterating :)

I replaced `new SparkSession` with `SparkSession.builder()` to create the session instance. In Spark 4.0, the builder API is kept in `sql.SparkSession`, and the classic SparkSession can be created with it by default.

In addition, I removed `SparkSession.cleanupAnyExistingSession` which requires using classic SparkSession companion object, then requires shim layer to differentiate the type if we call it. It's redundant to the existing code which already covers the cleanup.

@zhztheplayer, @zhouyuan, thanks for the review.
